tlatools/org.lamport.tlatools/src/tlc2/tool/liveness/TableauNodePtrTable.java

```diff
@@ -130,19 +132,33 @@ public final long get(final long k, final int tidx) {
 		}
 	}
 
+	public final void put(final long k, final int tidx) {
+		// Writing (long) UNDONE will result in node[3] = -2,
+		// and DONE will be node[3] = -3.  isDone is
+		// true if node[3] = -2.  Method setDone also sets
+		// node[3] = - 3 to mark the node as done.
+		put(k, tidx, UNDONE, DONE);
+	}
+	
 	/**
-	 * Add <tidx, elem> into the table. If the table has already contained <k,
-	 * tidx>, overwrite the old value.
+	 * Add the triple <k, tidx, elem> into the table. If the table already contains
+	 * and entry with the composite key <k, tdix>, the old elem will be replaced with
+	 * the new one.
 	 */
 	public final void put(long k, int tidx, long elem) {
+		put(k, tidx, elem, elem);
+	}
+
+	private final void put(long k, int tidx, long addElem, long newElem) {
+		assert tidx >= -1;
 		if (this.count >= this.thresh) {
 			this.grow();
 		}
 		int loc = ((int) k & 0x7FFFFFFF) % this.length;
 		while (true) {
 			int[] node = this.nodes[loc];
 			if (node == null) {
-				this.nodes[loc] = addElem(k, tidx, elem);
+				this.nodes[loc] = addElem(k, tidx, addElem);

```

Comment:
Should this be `newElem` instead of `addElem`? This branch seems most similar to the `appendElem` case below.

Reply:
That would violate four (newly created) tests:

tlc2.tool.liveness.TableauNodePtrTableTest.testIsDonePPS()
tlc2.tool.liveness.TableauNodePtrTableTest.testIsDonePSP()
tlc2.tool.liveness.TableauDiskGraphTest.testWhatsDoneIsDoneRRS()
tlc2.tool.liveness.TableauDiskGraphTest.testWhatsDoneIsDoneRSR()